File: crmuretek-frontend/src/types/Job.ts
export interface Job {
  id: number;
  estimateMaterialKg?: number;
  consulta?: {
    id: number;
    problemDescription?: string;
    customer: {
      id: number;
      name?: string;
      email?: string;
      phoneNumber?: string;
      address?: string;
      source?: string;
      contactDate?: string;
      title?: string | null;
      observations?: string | null;
    };
    requestDate?: string | null;
  };
  observations?: string;
  jobStatus: string;
  workDate?: string;
}

File: crmuretek/src/main/java/com/crmuretek/crmuretek/models/Job.java
    package com.crmuretek.crmuretek.models;

    import com.fasterxml.jackson.annotation.JsonManagedReference;
    import jakarta.persistence.*;
    import jakarta.validation.constraints.Min;
    import lombok.Data;

    import java.time.LocalDate;
    import java.util.List;

    @Entity
    @Table(name = "jobs")
    @Data
    public class Job {
        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        private Long id;

        @ManyToOne
        @JoinColumn(name = "leads_id")
        private Consulta consulta;

        @OneToOne
        @JoinColumn(name = "visit_id")
        private Visit visit;

        @OneToMany(mappedBy = "job", cascade = CascadeType.ALL, fetch = FetchType.EAGER)
        @JsonManagedReference("job-materials")
        private List<MaterialUsage> materialUsages;

        @Min(value = 0, message = "Material cannot be negative")
        private Double estimateMaterialKg; // estimation of kg to use in the job

        private String quoteNumber; // numero de presupesto aceptado


        @Column(columnDefinition = "TEXT")
        private String observations;

        @Enumerated(EnumType.STRING)
        private JobStatus jobStatus;     //status del trabajo
        private LocalDate workDate;   // fecha del trabajo



    }
